File: app/src/main/java/org/tvheadend/tvhclient/util/logging/BaseDebugTree.java
package org.tvheadend.tvhclient.util.logging;

import timber.log.Timber;

abstract class BaseDebugTree extends Timber.DebugTree {

    @Override
    protected String createStackElementTag(StackTraceElement element) {
        return String.format("[C:%s] [M:%s] [L:%s] ",
                super.createStackElementTag(element),
                element.getMethodName(),
                element.getLineNumber());
    }
}

File: app/src/main/java/org/tvheadend/tvhclient/util/logging/BaseDebugTree.kt
package org.tvheadend.tvhclient.util.logging

import timber.log.Timber

abstract class BaseDebugTree : Timber.DebugTree() {

    override fun createStackElementTag(element: StackTraceElement): String {
        return String.format("[C:%s] [M:%s] [L:%s] ",
                super.createStackElementTag(element),
                element.methodName,
                element.lineNumber)
    }
}
